library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -780,12 +779,13 @@ impl<T> VecDeque<T> {
     /// ```
     #[unstable(feature = "shrink_to", reason = "new API", issue = "56431")]
     pub fn shrink_to(&mut self, min_capacity: usize) {
-        assert!(self.capacity() >= min_capacity, "Tried to shrink to a larger capacity");
-
         // +1 since the ringbuffer always leaves one space empty
         // len + 1 can't overflow for an existing, well-formed ringbuffer.
```

Comment:
This comment already seems unhelpful pre this PR, as min_capacity could be greater than len. I guess maybe this should say something like `max(min(requested_capacity, capacity), len) + 1` can't overflow?

I admit to being a bit uncertain exactly what this logic is trying to aim for -- maybe @lcnr has some thoughts, IIRC you recently touched code in VecDeque?

Reply:
hmm, the comment doesn't seem too helpful. I do kind of see what it's meant to explain but it doesn't really help here so we can probably remove it.

Or replace it with 
```rust
// We don't have to worry about an overflow as neither `self.len()` nor `self.capacity()` can ever be `usize::MAX`.
```